Python/pyhash.c

```diff
@@ -168,17 +168,28 @@ _Py_HashBytes(const void *src, Py_ssize_t len)
         const unsigned char *p = src;
         hash = 5381; /* DJBX33A starts with 5381 */
 
-        switch(len) {
-            /* ((hash << 5) + hash) + *p == hash * 33 + *p */
-            case 7: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 6: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 5: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 4: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 3: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 2: hash = ((hash << 5) + hash) + *p++; _Py_FALLTHROUGH;
-            case 1: hash = ((hash << 5) + hash) + *p++; break;
-            default:
-                Py_UNREACHABLE();
+        if (len >= 4) {
+            hash = hash * 33 * 33 * 33 * 33 +
+                   p[0] * 33 * 33 * 33 +
+                   p[1] * 33 * 33 +
+                   p[2] * 33 +
+                   p[3];
```

Comment:
I think you can directly compute those values (namely the powers of 33), although the compiler should be able to optimize it as well. You could however add a small comment.

Reply:
Thank you for your review. Done.